src/app/pages/blog/blog-context.js

```diff
@@ -1,15 +1,94 @@
 import React from 'react';
 import {useHistory} from 'react-router-dom';
+import {useDataFromSlug} from '~/components/jsx-helpers/jsx-helpers.jsx';
 import buildContextLoader from '~/components/jsx-helpers/context-loader';
 import useLatestBlogEntries from '~/models/blog-entries';
+import cmsFetch from '~/models/cmsFetch';
+import $ from '~/helpers/$';
 
 const stayHere = {path: '/blog'};
 
+function useEnglishSubjects() {
+    const [data, setData] = React.useState([]);
+
+    React.useEffect(
+        () => cmsFetch('snippets/subjects?format=json&locale=en')
+            .then($.camelCaseKeys)
+            .then(setData),
+        []
+    );
+
+    return data;
+}
+
+function useCollections() {
+    const [data, setData] = React.useState([]);
+
+    React.useEffect(
+        () => cmsFetch('snippets/blogcollection?format=json')
+            .then($.camelCaseKeys)
+            .then(setData),
+        []
+    );
+
+    return data;
+}
+
+function useTopicStories() {
+    const [topicType, setTopicType] = React.useState();
+    const [topic, setTopic] = React.useState();
+    const setTypeAndTopic = React.useCallback(
+        (typ, top) => {
+            setTopicType(typ);
+            setTopic(top);
+        },
+        []
```

Comment:
I think `setTopic` and `setTopicType` should be listed in the dependencies array here, right?  I'm surprised that the linter didn't complain .